File: solutions/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPriceTillNow = prices[0];
        int maxProfit = 0;
        for(int i=0;i<prices.size();i++){
            if(minPriceTillNow>prices[i]){
                minPriceTillNow = prices[i];
            }
            maxProfit = max(maxProfit, prices[i]-minPriceTillNow);
        }
        return maxProfit;
    }
};

File: solutions/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minima = prices[0]
        maxprofit = 0
        for i in range(1,len(prices)):
            maxprofit = max(maxprofit, prices[i]-minima)
            minima = min(minima,prices[i])
        return maxprofit

